From bc56a1c5ea9029bd9dd4c2be8684e1ba27ba8f24 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 4 Jul 2024 10:04:20 +0200
Subject: [PATCH 1002/1695] drivers: stm32_omm: use firewall alternate
 configuration APIs

Drivers apply alternate configurations, use:
firewall_dt_get_alternate_conf() and firewall_set_alternate_conf()
to apply OSPI RIF configuration after driver management.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d244c5f9a382c4194d1d6b2f3f2cf97a5ed363a1 drivers: stm32_omm: use firewall alternate configuration APIs]
Change-Id: I5a85da011447695ed87f82d1b8ca80008bc182c9
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391366
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_omm.c | 20 +++++++-------------
 1 file changed, 7 insertions(+), 13 deletions(-)

diff --git a/core/drivers/stm32_omm.c b/core/drivers/stm32_omm.c
index 6504e5113..51a568f78 100644
--- a/core/drivers/stm32_omm.c
+++ b/core/drivers/stm32_omm.c
@@ -76,7 +76,7 @@ struct stm32_omm_pdata {
 	struct pinctrl_state *pinctrl_s;
 	struct stm32_ospi_pdata ospi_d[OSPI_NB];
 	struct stm32_mm_region region;
-	struct firewall_query *firewall_confs[OSPI_NB_RISUP];
+	struct firewall_alt_conf *firewall_conf;
 	vaddr_t base;
 	uint32_t mux;
 	uint32_t cssel_ovr;
@@ -128,12 +128,10 @@ static TEE_Result stm32_omm_parse_fdt(const void *fdt, int node)
 	if (res && res != TEE_ERROR_ITEM_NOT_FOUND)
 		return res;
 
-	for (i = 0; i < OSPI_NB_RISUP; i++) {
-		res = firewall_dt_get_by_index(fdt, node, i,
-					       &omm_d->firewall_confs[i]);
-		if (res)
-			return res;
-	}
+	res = firewall_dt_get_alternate_conf(fdt, node, "default",
+					     &omm_d->firewall_conf);
+	if (res)
+		return res;
 
 	omm_d->mux = fdt_read_uint32_default(fdt, node, "st,omm-mux", 0);
 	omm_d->req2ack = fdt_read_uint32_default(fdt, node,
@@ -328,17 +326,13 @@ static void stm32_omm_configure(void)
 
 static void stm32_omm_setup(void)
 {
-	unsigned int i = 0;
-
 	stm32_omm_set_mm();
 	stm32_omm_configure();
 	if (omm_d->pinctrl_d)
 		pinctrl_apply_state(omm_d->pinctrl_d);
 
-	for (i = 0; i < OSPI_NB_RISUP; i++) {
-		if (firewall_set_configuration(omm_d->firewall_confs[i]))
-			panic();
-	}
+	if (firewall_set_alternate_conf(omm_d->firewall_conf))
+		panic();
 }
 
 static void stm32_omm_suspend(void)
-- 
2.47.2

